blosc/blosc2.c

```diff
@@ -1258,10 +1261,10 @@ static int blosc_d(
     int32_t block_csize = block_csizes[nblock];
     // Read the lazy block on disk
     FILE* fp = NULL;
-    if (context->schunk->frame->eframe) {
+    if (frame->sframe) {
       // The chunk is not in the frame
-      char* chunkpath = malloc(strlen(context->schunk->frame->urlpath) + 1 + 8 + strlen(".chunk") + 1);
-      sprintf(chunkpath, "%s/%08X.chunk", context->schunk->frame->urlpath, nchunk);
+      char* chunkpath = malloc(strlen(frame->urlpath) + 1 + 8 + strlen(".chunk") + 1);
```

Comment:
I think there should be a helper function that opens chunk file, something like:
```
static inline FILE *frame_open_chunk_file(const char *urlpath, int32_t nchunk) {
  char* chunkpath = malloc(strlen(urlpath) + 1 + 8 + strlen(".chunk") + 1);
  sprintf(chunkpath, "%s/%08X.chunk", urlpath, nchunk);
  fp = fopen(chunkpath, "rb");
  free(chunkpath);
  return fp;
}
```
I noticed many places (in frame.c in general / not related to this PR) where `sprintf` is used in this manner and it could be encapsulated into helper functions.

Reply:
Ok.  Happy to see suggestions for helpers.